cosima_cookbook/database.py

```diff
@@ -192,6 +188,9 @@ def _setup_ncattribute(session, attr_object):
         if r is not None:
             return r
 
+    if attr_object.value in cache:
+        return cache[attr_object.value]
+
```

Comment:
@scottwales I'm just looking at squeezing some more performance out of the indexing and realised you made this change in this PR. Does't this stop the cache performing as a cache, in that it will always do a query whether or not that attribute has already been searched for?